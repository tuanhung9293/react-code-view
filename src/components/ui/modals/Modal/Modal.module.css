@value colors: 'styles/colors.css';
@value black, overlay, background-primary from colors;
@value shadows: 'styles/shadows.css';
@value heavy from shadows;

.modal{
  box-sizing: border-box;
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  min-height: 100%;
  overflow-y: auto;
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 70px 10px;
  z-index: 10000;
}

@media (max-width: 683px) {
  .modal{
    padding: 10px;
  }
}

.overlay{
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  margin: 0;
  z-index: 1000;
  background-color: overlay;
}

.notCenter{
  align-items: flex-start;
}

.content{
  z-index: 100000;
  background-color: background-primary;
  color: black;
  border-radius: 10px;
  box-shadow: heavy;
  max-width: 100%;
}

.transparent .content{
  background-color: transparent;
  box-shadow: none;
}