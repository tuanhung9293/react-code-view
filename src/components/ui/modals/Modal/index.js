import React from 'react'
import {createPortal} from 'react-dom'
import classNames from 'classnames'
import styles from './Modal.module.css'

const modalRoot = document.getElementById('modals')

class ModalPortal extends React.Component {
  constructor(props) {
    super(props)
    this.el = document.createElement('div')
  }
  componentDidMount() {
    modalRoot.appendChild(this.el)
  }
  componentWillUnmount() {
    modalRoot.removeChild(this.el)
  }
  render() {
    return createPortal(
      this.props.children,
      this.el
    )
  }
}

function Modal({children, notCenter, transparent}) {
  const className = classNames({
    [styles.modal]: true,
    [styles.notCenter]: notCenter,
    [styles.transparent]: transparent
  })
  return (
    <ModalPortal>
      <div className={styles.container}>
        <div className={styles.overlay} />
        <div className={className}>
          <div className={styles.content}>
            {children}
          </div>
        </div>
      </div>
    </ModalPortal>
  )
}

export default Modal