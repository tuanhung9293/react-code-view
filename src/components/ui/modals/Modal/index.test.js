import React from 'react'
import {shallow} from 'enzyme'
import Modal from './'

describe('test Modal', () => {
  it('renders Modal with no props', () => {
    expect(shallow(
      <Modal />
    )).toMatchSnapshot()
  })

  it('renders Modal with props notCenter', () => {
    expect(shallow(
      <Modal notCenter />
    )).toMatchSnapshot()
  })

  it('renders Modal with props transparent', () => {
    expect(shallow(
      <Modal transparent />
    )).toMatchSnapshot()
  })
})
