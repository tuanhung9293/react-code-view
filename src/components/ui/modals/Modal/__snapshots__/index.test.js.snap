// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Modal renders Modal with no props 1`] = `
<ModalPortal>
  <div
    className="container"
  >
    <div
      className="overlay"
    />
    <div
      className="modal"
    >
      <div
        className="content"
      />
    </div>
  </div>
</ModalPortal>
`;

exports[`test Modal renders Modal with props notCenter 1`] = `
<ModalPortal>
  <div
    className="container"
  >
    <div
      className="overlay"
    />
    <div
      className="modal notCenter"
    >
      <div
        className="content"
      />
    </div>
  </div>
</ModalPortal>
`;

exports[`test Modal renders Modal with props transparent 1`] = `
<ModalPortal>
  <div
    className="container"
  >
    <div
      className="overlay"
    />
    <div
      className="modal transparent"
    >
      <div
        className="content"
      />
    </div>
  </div>
</ModalPortal>
`;
